packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -83,7 +83,6 @@ export const filterRowTreeFromGroupingColumns = (
   params: FilterRowTreeFromTreeDataParams,
```

Comment:
@MBilalShafi 
>One way to solve this could be to only recompute the state.aggregation.lookup on demand
> ...
> This is definitely not an ultimate fix as it partially solves the problem of not re-computing the aggregation lookup on rows expansion change, but it causes another bug, i.e. now the grid is not responding properly to the aggregation function change.

We need to recompute the aggregation lookup on `filteredRowsSet`, even if aggregation rules remain unchanged.

My initial approach to solving this issue was to skip aggregation calculation if `filteredRowsLookup` has not changed. But then I realized that I can go a step ahead and skip both filtering and aggregation on row expansion/collapse.